src/content/docs/more-integrations/open-source-telemetry-integrations/opentelemetry/opentelemetry-tutorials/opentelemetry-tutorial-java.mdx

```diff
@@ -0,0 +1,1132 @@
+* Java 17 or higher: If you don't already have this, see [these download options](https://www.java.com/en/download/manual.jsp).
+
+## Run our pre-instrumented program [#pre-instrum]
+
+This is a great option if you want us do the instrumentation so you can quickly see what's it's like to send data to New Relic and view it in our UI.
+
+1. Clone the demo app and navigate to the following directory:
+    ```
+    git clone git@github.com:newrelic/newrelic-opentelemetry-examples.git
+    cd newrelic-opentelemetry-examples/getting-started-guides/java/Instrumented
+    ```
+2. In your terminal, set these two environment variables to send data to your New Relic account:
+    ```
+    export OTEL_EXPORTER_OTLP_HEADERS=api-key=INSERT_YOUR_NEW_RELIC_LICENSE_KEY
+    export OTEL_EXPORTER_OTLP_ENDPOINT=https://otlp.nr-data.net:4317
+
+    ```
+    * Make sure to use your [ingest license key](https://docs.newrelic.com/docs/apis/intro-apis/new-relic-api-keys/#license-key).
+    * If your account is based in the EU, set the endpoint to: https://otlp.eu01.nr-data.net:4317
+
+3. Set this environment variable to name the demo app:
+    ```
+    export OTEL_SERVICE_NAME=getting-started-java
+    ```
+4. In the `java/Instrumented/` directory, run the application:
+    ```
+    ./gradlew bootRun
+    ```
+5. Open a new terminal and exercise the application from the `Instrumented/java` directory by running this command:
+    ```
+    ./load-generator.sh
+    ```
+
+    <Callout variant="tip">
+      Alternatively, you can reach the endpoint in the browser at this URL: `http://localhost:8080/fibonacci?n=INSERT_A_VALUE`. Replace `INSERT_A_VALUE` with value from 1 to 90. To generate an error, insert an integer outside the valid range.
+    </Callout>
+
+6. Go to **[one.newrelic.com](https://one.newrelic.com/all-capabilities) > APM & services**.
+7. Click your new entity (service) called `getting-started-java` and explore the UI. For more tips about what to look for in the UI, see [View your data in New Relic](#view-data).
+8. When you're finished looking at your data in the UI, shut down the application by pressing CONTROL+C in both terminal sessions.
+
+## Use the OpenTelemetry Java agent [#java-agent]
+
+OpenTelemetry provides an agent that automatically instruments your Java apps and doesn't require you to  modify your source code. The agent is a quick way for you to get started so you can get sample data into New Relic.
+
+The auto-instrumentation agent is a JAR file that dynamically injects bytecode to capture telemetry from popular libraries and frameworks. You can also use it to capture data such as inbound requests, outbound HTTP calls, and database calls. It can be attached to any Java 8+ application.
+
+<Callout variant="tip">
+  See the [official documentation](https://opentelemetry.io/docs/instrumentation/java/automatic/) for additional configuration options.
```

Comment:
reminder about having descriptive url link text

Reply:
@zuluecho9 I revised it to this: See the [official Java agent documentation]....